### PR TITLE
[T3A3][T10-A1] Li Zhiyuan

### PR DESCRIPTION
**Pull request by lzyuan168**
ImplementClass

### CODE REVIEW

**Comment by lzyuan168** [comment]
ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,9 +8,14 @@
...
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses to be in this format";
+    public static final String ADDRESS_VALIDATION_REGEX = ",+";
+    
+    private final Block block;
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java]: Why is this final? Do you have setter methods to change these variables, if a value change? Check out the implications of making a variable final

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -26,10 +31,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        String[] fullAddress = address.split(", ");
+        this.block = new Block(fullAddress[0].trim());
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java]: - What if all 4 component are not entered by the user?

- Did you run the regression? It is a must to run the regression and change testcases if needed with these changes.

**Comment by nirandiw** [comment]
Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**